soem/ethercatcoe.c

```diff
@@ -404,13 +404,13 @@ int ecx_SDOwrite(ecx_contextt *context, uint16 Slave, uint16 Index, uint8 SubInd
          framedatasize = maxdata;  /*  segmented transfer needed  */
          NotLast = TRUE;
       }
-      SDOp->MbxHeader.length = htoes(0x0a + framedatasize);
+      SDOp->MbxHeader.length = (uint16)htoes(0x0a + framedatasize);
```

Comment:
(uint16) cast necessary?

Reply:
Yes, but as in previous comment, if I revert and make framedatasize an uint16 it is not needed. Gien I also make maxdata an uint16, 
And in addition I do explicit casts of API argument int pzise to uint16,

Does that make sense.

Reply:
Maxdata can be larger than uint16. The issue I have is mostly the (uint16)htoes(). htoes() is a short. The only reason llvm complains is because htoes() is a macro and does not convert int to uint16 itself. I would leave maxdata and framesize as before and then simply do : SDOp->MbxHeader.length= htoes((uint16)(0x0a + framedatasize)); This way the intend is much more clear.

Reply:
fixed